src/fmt.ml

```diff
@@ -187,6 +210,23 @@ let styled_string style = styled style string
 let stdout = Format.std_formatter
 let stderr = Format.err_formatter
 
+let with_strfmt f =
+  let buf = Buffer.create 17 in
+  let ppf = Format.formatter_of_buffer buf in
+  f ppf ;
+  Format.pp_print_flush ppf () ;
+  Buffer.contents buf
+
+let strfmt pp_v v = strf "%a" pp_v v
+
+let with_file filename f =
```

Comment:
I don't like this, as it doesn't handle errors cleanly and can leak fds, raise exceptions etc.. I don't think I want to make `Fmt` dependent on `rresult`. Maybe this is more something that belongs to [`Bos.Os.File`](http://erratique.ch/software/bos/doc/Bos.OS.File.html).